tools/src/main/java/org/apache/kafka/trogdor/common/WorkerUtils.java

```diff
@@ -85,58 +88,130 @@ public static int perSecToPerPeriod(float perSec, long periodMs) {
      */
     public static void createTopics(Logger log, String bootstrapServers,
             Collection<NewTopic> topics) throws Throwable {
-        Properties props = new Properties();
-        props.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
-        props.put(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG, CREATE_TOPICS_REQUEST_TIMEOUT);
-        try (AdminClient adminClient = AdminClient.create(props)) {
-            long startMs = Time.SYSTEM.milliseconds();
-            int tries = 0;
+        Collection<String> topicsExists;
+        try (AdminClient adminClient = createAdminClient(log, bootstrapServers)) {
+            topicsExists = createTopics(log, adminClient, topics);
+        }
+        if ((topicsExists != null) && !topicsExists.isEmpty()) {
+            log.warn("Topic(s) {} already exist.", topicsExists);
+            throw new TopicExistsException("One or more topics already exist.");
+        }
+    }
 
-            Map<String, NewTopic> newTopics = new HashMap<>();
-            for (NewTopic newTopic : topics) {
-                newTopics.put(newTopic.name(), newTopic);
-            }
-            List<String> topicsToCreate = new ArrayList<>(newTopics.keySet());
-            while (true) {
-                log.info("Attemping to create {} topics (try {})...", topicsToCreate.size(), ++tries);
-                Map<String, Future<Void>> creations = new HashMap<>();
-                while (!topicsToCreate.isEmpty()) {
-                    List<NewTopic> newTopicsBatch = new ArrayList<>();
-                    for (int i = 0; (i < MAX_CREATE_TOPICS_BATCH_SIZE) &&
-                            !topicsToCreate.isEmpty(); i++) {
-                        String topicName = topicsToCreate.remove(0);
-                        newTopicsBatch.add(newTopics.get(topicName));
-                    }
-                    creations.putAll(adminClient.createTopics(newTopicsBatch).values());
+    static Collection<String> createTopics(Logger log, AdminClient adminClient,
```

Comment:
I understand why you did it this way, but it's a little counter-intuitive that this returns a collection containing topics that already exist.  I wouldn't expect that just by reading the function signature.  Probably we should have a JavaDoc comment explaining the return type.  Or possibly once `createTopics` and `verifyTopicsAndCreateNonExistingTopics` are merged, we won't need this function.